src/User.cpp

```diff
@@ -10,13 +10,15 @@ User::User(const std::string &name,
            const std::string &firecode,
            const std::string &email,
            const std::string &password)
-    : name_(name)
+    : hash_(PasswordHashingProvider(16,16))
```

Comment:
If we would like to test user without this hash_ how can this be achieved?
Here we have tight coupling.

Reply:
I've considered making one instance of my class and passing it through c-tor of user class, but then I'd have to refactor a huge amount of existing unit tests...
I could really use a hint on this :)

Reply:
You should create an interface IPasswordHashingProvider. Then you can create a mock/dummy/stub of this inteface, which can be used in UserTests